syntax = "proto3";

package santiago.web.localization;

service ClientLocalization {
    rpc getTranslations (GetTranslationsRequest) returns (GetTranslationsResponse) {}
}

message SidList {
    repeated string sids = 1;
}

message CategoryIdList {
    repeated int32 ids = 1;
}

message SubCategoryIdList {
    repeated int32 ids = 1;
}

message TranslationsQuery {
    optional SidList sidList = 1;
    optional CategoryIdList categoryIdList = 2;
    optional SubCategoryIdList subCategoryIdList = 3;
}

message LocalizedText {
    string sid = 1;
    string localizedText = 2;
    int32 categoryId = 3;
}

message Timestamp {
    int64 seconds = 1 [jstype=JS_STRING];
    int32 nanos = 2;
}

message GetTranslationsRequest {
    TranslationsQuery translationsQuery = 1;
    string locale = 4;
    optional Timestamp fetchFrom = 5;
}

message GetTranslationsResponse {
    repeated LocalizedText localizedTexts = 3;
    optional Timestamp fetchedUntil = 4;
}
